src/pages/index.js

```diff
@@ -56,31 +68,87 @@ function handleImageClick(data) {
 }
 
 function handleEditProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    job: inputValues.description,
-  });
-  editProfileModal.close();
+  const { title, description } = inputValues;
+  editProfileModal.setSaving(true);
+  api
+    .updateProfile(title, description)
+    .then((data) => {
+      userInfo.setUserInfo({
+        name: data.name,
+        job: data.about,
+      });
+      editProfileModal.close();
+    })
+    .catch(handleError)
+    .finally(() => editProfileModal.setSaving(false));
+}
+
+function handleDeleteCard(card) {
+  {
+    confirmPopup.setSubmitFunction(() => {
+      confirmPopup.setSaving(true);
+      api
+        .deleteCard(card._id)
+        .then(() => {
+          card.remove();
+          confirmPopup.setSaving(false);
+          confirmPopup.close();
+        })
+        .catch(handleError);
+    });
+
+    confirmPopup.open();
+  }
 }
 
 function handleAddCardFormSubmit(inputValues) {
+  addCardModal.setSaving(true);
   const cardData = { name: inputValues.title, link: inputValues.url };
-  const newCard = createCard(cardData);
-  cardSection.addItem(newCard);
-  addCardModal.close();
-  modalAddForm.reset(); // clear the form
-  addCardFormValidator.disableButton(); // disable the button
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      addCardModal.close();
+      modalAddForm.reset();

```

Comment:
`  modalAddForm.reset();`


* When I add a new card, there is an error https://snipboard.io/1ZqVv6.jpg 

`modalAddForm` doesn't exist. You have `newCardEditForm` as the card form


```js

      newCardEditForm.reset();

```